modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/senna/screen/RequestScreen.js

```diff
@@ -233,52 +240,65 @@ class RequestScreen extends Screen {
 		}
 		let body = null;
 		let httpMethod = this.httpMethod;
-		const headers = {...this.httpHeaders};
+		const requestHeaders = {'X-PJAX': 'true', ...this.httpHeaders};
 		if (globals.capturedFormElement) {
-			this.addSafariXHRPolyfill();
 			body = this.getFormData(
 				globals.capturedFormElement,
 				globals.capturedFormButtonElement
 			);
 			httpMethod = RequestScreen.POST;
 		}
-		const requestPath = this.formatLoadPath(path);
-
-		return Ajax.request(
-			requestPath,
-			httpMethod,
-			body,
-			headers,
-			null,
-			this.timeout
-		)
-			.then((xhr) => {
-				this.removeSafariXHRPolyfill();
-				this.setRequest(xhr);
-				this.assertValidResponseStatusCode(xhr.status);
-				if (httpMethod === RequestScreen.GET && this.isCacheable()) {
-					this.addCache(xhr.responseText);
-				}
-				xhr.requestPath = requestPath;
 
-				return xhr.responseText;
+		const url = this.formatLoadPath(path);
+
+		this.setRequest({
+			method: httpMethod,
+			requestBody: body,
+			requestHeaders,
+			url
+		});
+
+		return Promise.race([
+			fetch(url, {
+				body,
+				headers: requestHeaders,
+				method: httpMethod,
+				mode: 'cors',
+				redirect: 'follow',
+				referrer: 'client'
 			})
-			.catch((reason) => {
-				this.removeSafariXHRPolyfill();
-				switch (reason.message) {
-					case errors.REQUEST_TIMEOUT:
-						reason.timeout = true;
-						break;
-					case errors.REQUEST_ERROR:
-						reason.requestError = true;
-						break;
-					case errors.REQUEST_PREMATURE_TERMINATION:
-						reason.requestError = true;
-						reason.requestPrematureTermination = true;
-						break;
-				}
-				throw reason;
-			});
+				.then(resp => {
+					this.assertValidResponseStatusCode(resp.status);
+
+					this.setResponse(resp);
+
+					return resp.clone().text();
+				})
+				.then(text => {
+					if (httpMethod === RequestScreen.GET && this.isCacheable()) {
+						this.addCache(text);
+					}
+
+					return text;
+				}),
+			new Promise((_, reject) => {
+				setTimeout(() => reject(new Error(errors.REQUEST_TIMEOUT)) , this.timeout);
+			})
```

Comment:
The formatting makes this hard to read and I can see a space before the comma on line 285. Would this be better?

```js
const timeout = new Promise((_, reject) => {
    setTimeout(() => reject(new Error(errors.REQUEST_TIMEOUT)), this.timeout);
});

return Promise.race([
    timeout,
    fetch(/*
       etc
    /*)
        .then()
        // etc
);